pkg/config/config.go

```diff
@@ -268,7 +268,7 @@ func (m *Manager) ReadInConfig() error {
 
 		m.viper.SetConfigFile(file)
 		err := m.viper.MergeInConfig()
-		if err != nil {
+		if err != nil && m.inCLIFlags(file) {
```

Comment:
I think it would be better to a check for `os.IsPermission` on L264 (and update the comment above).

If files can't be read for another reason (for instance invalid yaml), we should error here, also if those files are in an implicit/default location